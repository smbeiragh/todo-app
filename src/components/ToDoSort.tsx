import * as React from 'react';
import classnames from 'classnames';

type Props = {
  onChange: (value: string) => void
  value: string
  className: string
}

type DefaultProps = {
  onChange: (value: string) => void
  value: string
  className: string
}

export class ToDoSort extends React.Component<Props> {

  static defaultProps: DefaultProps = {
    onChange: () => {},
    value: '',
    className: ''
  }

  constructor(props: Props) {
    super(props)
  }

  handleChange(value: string) {
    if (this.props.value === value) {
      this.props.onChange('');
    } else {
      this.props.onChange(value);
    }
  }

  render() {
    const { className, value } = this.props;
    return (
      <div className={classnames(className, 'c-toDoSort')}>
        <div className="c-toDoSort__label">Sort By</div>
        <button
          className={classnames('c-toDoSort__btn', {['is-active']: value === 'title'})}
          onClick={ (e) => { e.preventDefault(); this.handleChange('title'); } }
        >
          Name
        </button>
        <button
          className={classnames('c-toDoSort__btn', {['is-active']: value === 'dueDate'})}
          onClick={ (e) => { e.preventDefault(); this.handleChange('dueDate'); } }
        >
          Due Date
        </button>
        <button
          className={classnames('c-toDoSort__btn', {['is-active']: value === 'priority'})}
          onClick={ (e) => { e.preventDefault(); this.handleChange('priority'); } }
        >
          Priority
        </button>
      </div>
    );
  }
}
