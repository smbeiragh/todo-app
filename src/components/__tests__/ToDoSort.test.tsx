import * as React from 'react';
import { ToDoSort } from '../ToDoSort';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';


describe('ToDoSort', () => {
  it('click on sort buttons should trigger onChange', () => {
    const clickFn = jest.fn();
    const component = shallow(<ToDoSort onChange={clickFn} />);
    component
      .find('button.c-toDoSort__btn')
      .at(0)
      .simulate('click', {
        preventDefault: () => {
        }
      });
    expect(clickFn).toHaveBeenCalled();
  });

  it('ToDoSort should render', () => {
    const component = renderer.create(
     <ToDoSort />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
