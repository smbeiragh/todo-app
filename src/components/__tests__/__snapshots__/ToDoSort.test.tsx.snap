// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoSort ToDoSort should render 1`] = `
<div
  className="c-toDoSort"
>
  <div
    className="c-toDoSort__label"
  >
    Sort By
  </div>
  <button
    className="c-toDoSort__btn"
    onClick={[Function]}
  >
    Name
  </button>
  <button
    className="c-toDoSort__btn"
    onClick={[Function]}
  >
    Due Date
  </button>
  <button
    className="c-toDoSort__btn"
    onClick={[Function]}
  >
    Priority
  </button>
</div>
`;
